core/node/rules/can_create_stream.go

```diff
@@ -484,13 +484,14 @@ func (ru *csChannelRules) derivedChannelSpaceParentEvent() (*DerivedEvent, error
 	}
 
 	payload := events.Make_SpacePayload_ChannelUpdate(
```

Comment:
it looks like Evan might not get to this work this week, it's not his highest priority, so we need to make sure it's backwards compatable with old clients by checking for a nil ChannelSettings here, and if ChannelSettings is nil, adding an autoJoin=true if the channelId is a default channel id?

Reply:
whoa good catch

Reply:
Well actually, I think the snapshot will render the channel setting as autojoin=true when the channelupdate is emitted with empty channel settings for the default channel... but if you'd prefer this change, no problem.

Reply:
Oh wow - yeah looks like that would happen - can we make this change anyway in case that code changes or we start looking at the snapshot in a different way in the future?

Reply:
actually never mind, lets leave this how it is

Reply:
Oh I went ahead and made it, but I left the old one there too